src/engine/strat_engine/liminal/device_info.rs

```diff
@@ -410,6 +410,46 @@ impl DeviceSet {
             .next()
     }
 
+    /// The encryption information and devices registered for this locked pools to be
+    /// exported over the API. If none of the infos correspond to a Stratis managed
+    /// encrypted device, None.
+    pub fn locked_pool_info(&self) -> Option<LockedPoolInfo> {
+        let encryption_info = self
+            .internal
+            .iter()
+            .filter_map(|(_, info)| info.encryption_info())
+            .next();
+        encryption_info.and_then(|info| {
+            let devices = self
+                .internal
+                .iter()
+                .map(|(uuid, l)| {
+                    let devnode = match l {
+                        LInfo::Stratis(strat_info) => {
+                            strat_info.luks.as_ref().map(|l| l.ids.devnode.clone())
+                        }
+                        LInfo::Luks(luks_info) => Some(luks_info.ids.devnode.clone()),
+                    };
+                    devnode.map(|devnode| LockedPoolDevice {
+                        devnode,
+                        uuid: *uuid,
+                    })
+                })
+                .fold(Some(Vec::new()), |vec, dev_info| {
```

Comment:
I think this fold returns None if the LUKS device node was not known for any of the devices belonging to the set. Could you add that information in the header comment?

Reply:
I'm not sure I entirely agree. The devnode will always be known if it is an encrypted Stratis device. What it's checking here is whether or not it's encrypted, not whether or not it knows the device node. The device node is not optional, the encryption status is.

Reply:
I think we are saying the same thing. One of the devices belonging to the set could be represented as a Stratis device w/out an underlying LUKS device. Unlikely but possible if the LUKS device hasn't been picked up via udev. In that case, you would not return this set as part of the result. This seems to me like a reasonable decision, but it should be documented, because it is not obvious without mulling over the ```fold``` fairly carefully.

Reply:
Okay, I understand now! I'll add a comment.